File: backend/src/types/presentation.ts
export type Slide = {
  title: string;
  content: string[];
  imageUrl?: string;
};

export type PresentationOptions = {
  theme?: string;
  slideCount?: number;
  includeImages?: boolean;
};

export type Presentation = {
  id: string;
  slides: Slide[];
  theme: string;
  createdAt: string;
  downloadUrl: string;
};

File: backend-python/app/schemas/presentation.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime

class Slide(BaseModel):
    title: str
    content: List[str]
    image_url: Optional[str] = None

class PresentationOptions(BaseModel):
    theme: Optional[str] = "modern"
    slide_count: Optional[int] = 10
    include_images: Optional[bool] = True

class PresentationRequest(BaseModel):
    text: str
    options: Optional[PresentationOptions] = Field(default_factory=PresentationOptions)

class Presentation(BaseModel):
    id: str
    slides: List[Slide]
    theme: str
    created_at: str
    download_url: str
